### ISSUE TITLE
[Bug]: Does not trigger build with esbonio 0.11.0

### ISSUE THREAD

**Issue by twodrops** [vscode-restructuredtext/vscode-restructuredtext#384]
### Expected behavior

I expected that the build generation works as with esbonio 0.10.3.

That is,

> updating environment:
[new config]
n added, n changed, n removed
reading sources...

### Actual behavior

Build does not proceed after

> updating environment:
[new config]
n added, n changed, n removed

Does not proceed to
> reading sources...

### Detailed steps

* Open VsCode
* Observe the VsCode Output View

### Extension information

185.0.0/184.0.0

### Visual Studio Code information

1.65.1

### Operating system information

Ubuntu 20.04

### Python information

3.8.10

### Sphinx information

4.5.0

**Comment by twodrops** [comment]
How can I increase the log level of ebsonio (`server.logLevel`) to `info` or `debug` when using the rST Extension?

**Comment by alcarney** [comment]
Setting `esbonio.server.logLevel = debug` should be enough

**Comment by twodrops** [comment]
@alcarney I tried setting it in `settings.json`, but it appears to be greyed out. 

![image](https://user-images.githubusercontent.com/103930/164365207-8caddbd6-2e65-4e21-9bef-1830e4c1a07d.png)

Btw, I have only the rST VsCode Extn installed and not the Esbonio VsCode Extn. Am I doing something wrong here?

**Comment by twodrops** [comment]
@alcarney Does esbonio 0.11.0 work for you with rST Extn 185.0.0/184.0.0?

**Comment by lextm** [comment]
@twodrops `esbonio.server.logLevel` was displayed as grey because in previous release that setting's scope was `application`. You can leave it that way, as the extension forwards that setting to the language server no matter what.

BTW, release 186.0.0 moves that setting to `window` scope so it won't grey out. But that's less relevant for troubleshooting. We do need to see the actual debug log entries so as to learn what prevents "reading sources.." from printing.

**Comment by twodrops** [comment]
@lextm  ah Ok. Thanks for the feedback.

I will post the debug info soon.

Just to make myself clear. The problem is not that  "reading sources.."  is not being printed. The read and write does not happen anymore. If I switch back to 0.10.3 everything works fine.

**Comment by oharboe** [comment]
I'm also not able to get preview to work with 0.11.0

**Comment by alcarney** [comment]
@twodrops I've tried on both Windows and Linux, and the combination of vscode-restructuredtext v185.0.0 and esbonio 0.11.0 works fine for me (including preview)...

@oharboe are there any logs that might hint at what is happening?

**Comment by twodrops** [comment]
I tested it again (rST 186.0.0, esbonio 0.11.0) and it seems to be a problem with the new parallel build feature in esbonio.

If I leave all settings as default, all 5 cores I have seems to be used and then it stops at read.
```
[esbonio.lsp] User Config {
  "version": null,
  "conf_dir": "${workspaceFolder}/docs",
  "src_dir": null,
  "build_dir": "${workspaceFolder}/docs/_build",
  "builder_name": "html",
  "force_full_build": true,
  "num_jobs": 5
}
[esbonio.lsp] Sphinx Args {
  "buildername": "html",
  "confdir": "/workspaces/docs",
  "doctreedir": "/workspaces/docs/_build/doctrees",
  "freshenv": true,
  "outdir": "/workspaces/docs/_build/html",
  "parallel": 5,
  "srcdir": "/workspaces/docs",
  "status": null,
  "warning": null
}
```
If I set it to a single core, everything works again.
`"esbonio.sphinx.numJobs":1`

```
[esbonio.lsp] User Config {
  "version": null,
  "conf_dir": "${workspaceFolder}/docs",
  "src_dir": null,
  "build_dir": "${workspaceFolder}/docs/_build",
  "builder_name": "html",
  "force_full_build": true,
  "num_jobs": 1
}
[esbonio.lsp] Sphinx Args {
  "buildername": "html",
  "confdir": "/workspaces/docs",
  "doctreedir": "/workspaces/docs/_build/doctrees",
  "freshenv": true,
  "outdir": "/workspaces/docs/_build/html",
  "parallel": 1,
  "srcdir": "/workspaces/docs",
  "status": null,
  "warning": null
}
```
Btw, I am testing this in a VsCode Devcontainer with Ubuntu 20.04 where a `sphinx-build -j auto` works fine.

Irrespective of the problem above, shall we leave the default parallel build as off?

**Comment by alcarney** [comment]
I'm struggling to imagine why `sphinx.numJobs` would be causing an issue, as it boils down to passing `parallel = N` to the Sphinx app object - just like the `-j auto` cmd line argument does...

However, I'm also unable to reproduce this so I'm not able to debug it.

- Do you see still this behavior in a default `sphinx-quickstart` project?
- Does the language server still function (e.g. completions) despite the build not working, or does the server effectively hang indefinitely?

**Comment by lextm** [comment]
Shipped 187.0.0 with default value changed to 1 for `sphinx.numJobs`.

This should give us more time to investigate why `0` doesn't work out.
